internal/worker/pool.go

```diff
@@ -0,0 +1,78 @@
+package worker
+
+import (
+	"context"
+	"log/slog"
+
+	"github.com/supressionstop/xenking_test_1/internal/config"
+	"github.com/supressionstop/xenking_test_1/internal/usecase"
+)
+
+type Pool struct {
+	workers        []*Worker
+	logger         *slog.Logger
+	FirstSyncChan  chan struct{}
+	workerSyncChan chan struct{}
+	workersErrChan chan error
+	ErrChan        chan error
+}
+
+func NewPool(cfg config.Config, logger *slog.Logger, fetchLine usecase.FetchLine) *Pool {
```

Comment:
From my understanding, this package can be a pure implementation for the worker pool without any domain logic leaks. I understand that in this implementation it's OK.

Reply:
TODO: try to replace many structs to worker pool pattern.

Reply:
Need to discuss new solution.